src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -331,6 +331,27 @@ public function get_items( $request ) {
 			if ( ! current_user_can( 'list_users' ) ) {
 				$prepared_args['search_columns'] = array( 'ID', 'user_login', 'user_nicename', 'display_name' );
 			}
+			$search_columns         = $request->get_param( 'search_columns' );
+			$valid_columns          = isset( $prepared_args['search_columns'] )
```

Comment:
I think we should change the check here to `! current_user_can( 'list_users' )`. In essence remove the above check and bake it here.